packages/output/src/browser/output-widget.tsx

```diff
@@ -93,13 +94,16 @@ export class OutputWidget extends BaseWidget implements StatefulWidget {
         this.onStateChangedEmitter.fire(this._state);
     }
 
-    protected async refreshEditorWidget(): Promise<void> {
+    protected async refreshEditorWidget({ preserveFocus }: { preserveFocus: boolean } = { preserveFocus: false }): Promise<void> {
         const { selectedChannel } = this;
-        const editor = this.editor;
-        if (selectedChannel && editor) {
+        const editorWidget = this.editorWidget;
+        if (selectedChannel && editorWidget) {
             // If the input is the current one, do nothing.
-            const model = editor.getControl().getModel();
+            const model = (editorWidget.editor as MonacoEditor).getControl().getModel();
             if (model && model.uri.toString() === selectedChannel.uri.toString()) {
+                if (!preserveFocus) {
+                    MessageLoop.sendMessage(editorWidget, Widget.Msg.ActivateRequest);
```

Comment:
usually activation is done via `ApplicationShell.activate`. It makes sure to activate all widgets in parent-child order.

It is strange that the widget itself reacts and try to update visibility/focus without knowing visibility/activation state of its parents.

Reply:
> activate all widgets in parent-child order.

Hmm, that's interesting. We use the `MessageLoop.sendMessage(myWidget, Widget.Msg.ActivateRequest);` construct overalll the application:
 - view container,
 - terminal,
 - scm,
 - siw,
 - editor preview.

What's wrong with this one?

Reply:
I see; you're right. Let me check why it was needed. Thanks for the heads-up.

Reply:
Are you sure? I think it is the only place where we use it.

Reply:
Yes, you're right.

Reply:
Widget should implement how it is activated in `onActivateRequest`, but activation is done via `ApplicationShell.activate`. It will analyze what are parents (areas, tabs, composite widgets) of widgets and reveal/activate them first after that it will ask the widget to activate.

Reply:
It does not seem to be caused by your changes though, so probably out of the scope to refactor it. I guess it works because there is another path running in parallel making sure that the output view is visible and activated.